### ISSUE TITLE
RFE: GitHub login/authentication integration

### ISSUE THREAD

**Issue by TPS** [vfeskov/win-a-beer#11]
If 1's GitHub account were integrated into Win-A-Beer, some things might be possible:

- [x] Single sign-on, so don't have to remember yet another password
- [ ] Import already contributed, watched, ★ed, &c, repos

What do you think, @vfeskov?  Is it possible?

**Comment by vfeskov** [comment]
good idea, i'll work on that

**Comment by vfeskov** [comment]
@TPS  first step done! <img width="698" alt="screen shot 2018-01-11 at 23 31 19" src="https://user-images.githubusercontent.com/2743367/34850781-94e67aca-f727-11e7-8a2c-709bf544b11a.png">

**Comment by TPS** [comment]
Noted above, but the UI doesn't update when signed in, so it's not really clear that it worked.…

**Comment by vfeskov** [comment]
You can test it by signing in in incognito tab or in different browser. I will also add github integration option to signed in users soon, and will show some indicator if they already integrated

**Comment by vfeskov** [comment]
@TPS 
<img width="286" alt="screen shot 2018-01-12 at 15 57 47" src="https://user-images.githubusercontent.com/2743367/34880648-637720a4-f7b1-11e7-99a2-4449391cd570.png">
<img width="241" alt="screen shot 2018-01-12 at 15 57 57" src="https://user-images.githubusercontent.com/2743367/34880647-635c6d04-f7b1-11e7-9eee-a7af0e29427b.png">

if you were signed out and clicked on **SIGN IN WITH GITHUB** it would connect your existing account to the github one, but this way is more straightforward.

having github connected will provide many benefits in future.

**Comment by TPS** [comment]
@vfeskov Sorry for disappearing for so long.  Life happened.…

Anyway, what pleasant updates since I last checked!  _Almost_ everything I asked for, but I'd like to request an additional "Select/Unselect All" option to this UI.

**Comment by vfeskov** [comment]
@TPS hi, can you please open another issue for "select all" ? i think it's a good idea, especially since i refactored code to be much more scalable then before

i have a question about watched repos, don't watchers get emails about releases anyway?

with repos contributed to there's no official github api AFAIK and solutions are pretty [sketchy](https://stackoverflow.com/questions/20714593/github-api-repositories-contributed-to), like pulling latest 100 commits or something. Do you have many repos you contributed to but didn't star?

**Comment by TPS** [comment]
Watchers get a _flood_ of _everything, so a separate release notification would be easier to find/notice/sort.  As for each option, if 1 has, say, >1k ★ed repos, maybe separate UI to see just watched/contributed repos would be useful.  Have you seen Sibbell.com?

**Comment by vfeskov** [comment]
i don't know it feels like when you decide to watch you're not supposed to just ignore emails. Like if I was in such shoes I would make an explicit email filter for notifications that I want to ignore, and obviously I wouldn't ignore release notifications

Is your use-case like this? do you watch repos but need a separate sender specifically for release emails?

Sibbell.com is nice and if i knew about it before starting Win A Beer I probably wouldn't start it. But at this point I got some things here that make Win a Beer stand out, and hopefully not having questionable features is going to be one of them. For example, Win a Beer is open-source, daily notifications are offered for free and you can star whatever you want without being punished by unwanted release emails. 

I'll continue to add more features, but i wouldn't add them just for the sake of copying sibbell.com or others

**Comment by TPS** [comment]
A way to _filter_ my e-mails better for releases works, so a separate sender helps, yes.

Open-source is _precisely_ why I'm trying to make Win A Beer better.  I just mentioned Sibbell as an example of they show different panels for ★ed/Watched/&c.  Other than that, Win A Beer's features _are_ superior, for a better price (of course 😀).
